interpreter/llvm/src/tools/clang/lib/AST/ODRHash.cpp

```diff
@@ -136,22 +136,8 @@ void ODRHash::AddNestedNameSpecifier(const NestedNameSpecifier *NNS) {
 }
 
 void ODRHash::AddTemplateName(TemplateName Name) {
-  auto Kind = Name.getKind();
-  ID.AddInteger(Kind);
-
-  switch (Kind) {
-  case TemplateName::Template:
-    AddDecl(Name.getAsTemplateDecl());
-    break;
-  // TODO: Support these cases.
-  case TemplateName::OverloadedTemplate:
-  case TemplateName::AssumedTemplate:
-  case TemplateName::QualifiedTemplate:
-  case TemplateName::DependentTemplate:
-  case TemplateName::SubstTemplateTemplateParm:
-  case TemplateName::SubstTemplateTemplateParmPack:
-    break;
-  }
+  if (auto *TD = Name.getAsTemplateDecl())
```

Comment:
Is it worth to try out if https://reviews.llvm.org/D141224 fixes it? That should be upstream already.

Reply:
The changes from https://reviews.llvm.org/D141224 are included in our application of D41416 (where they have traditionally been part of).

Also, this is a backport for 6.28. I will only cherry-pick what is already in `master`.

Reply:
So, besides https://reviews.llvm.org/D141224 we still need that change. Ok, weird.

Reply:
Note that these are completely different methods, `AddTemplateName` is called in case of `TemplateArgument::Template`.

Reply:
I was wondering if we could use the same approach but we can move the discussion under the phab review.